sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/ItemsScan.scala

```diff
@@ -26,9 +29,23 @@ private case class ItemsScan(schema: StructType,
   }
 
   override def planInputPartitions(): Array[InputPartition] = {
-    // TODO: moderakh use get feed range?
-    // for now we are returning one partition hence only one spark task will be created.
-    Array(FeedRangeInputPartition(FeedRange.forFullRange.toString))
+    val readConfig = CosmosReadConfig.parseCosmosReadConfig(config)
+    val clientConfiguration = CosmosClientConfiguration.apply(config, readConfig.forceEventualConsistency)
+    val containerConfig = CosmosContainerConfig.parseCosmosContainerConfig(config)
+    val partitioningConfig = CosmosPartitioningConfig.parseCosmosPartitioningConfig(config)
+    val defaultMaxPartitionSizeInMB = (session.sessionState.conf.filesMaxPartitionBytes / (1024 * 1024)).toInt
+
+    val defaultMinPartitionCount = 1 + (2 * session.sparkContext.defaultParallelism)
+
+    CosmosPartitionPlanner.createInputPartitions(
+      clientConfiguration,
```

Comment:
how do we want to deal with the fact that filters might be pushed down? My rough idea was if no filters are pushed down, use default partitioning strategy -(storage based) but use restrictive if any filters are pushed down./ bot as the default if the customer didn't specify in config

Reply:
I created a work item for this. I will cover in another PR if that's fine with you.
https://github.com/Azure/azure-sdk-for-java/issues/19613